app/src/main/java/org/sagebionetworks/research/mpower/tracking/fragment/DurationFragment.java

```diff
@@ -0,0 +1,186 @@
+package org.sagebionetworks.research.mpower.tracking.fragment;
+
+import android.os.Bundle;
+import android.support.annotation.ColorRes;
+import android.support.annotation.NonNull;
+import android.support.v4.view.OnApplyWindowInsetsListener;
+import android.support.v4.view.ViewCompat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper;
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper.Direction;
+import org.sagebionetworks.research.mobile_ui.widget.ActionButton;
+import org.sagebionetworks.research.mpower.R;
+import org.sagebionetworks.research.mpower.tracking.model.TrackingItem;
+import org.sagebionetworks.research.mpower.tracking.view_model.SymptomTrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.view_model.configs.SimpleTrackingItemConfig;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.SymptomLog;
+import org.sagebionetworks.research.presentation.model.interfaces.StepView;
+import org.threeten.bp.Instant;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.BindViews;
+
+/**
+ * The DurationFragment allows the user to select from the duration options to specify how long a symptom has occurred for.
+ * It is an invariant that there is a log in the view model for the TrackingItem provided to this fragment as an argument,
+ * and the client of this fragment is responsible for ensuring this is correct.
+ */
+public class DurationFragment extends TrackingFragment<SimpleTrackingItemConfig, SymptomLog, SymptomTrackingTaskViewModel> {
+    public static final String ARGUMENT_TRACKING_ITEM = "trackingItem";
+    public static final String FORWARD_BUTTON_TEXT = "Save";
+
+    private TrackingItem trackingItem;
+    @ColorRes
+    private static final int SELECTED_COLOR = R.color.royal200;
```

Comment:
Shouldn't this be done automatically using the color themes?

Reply:
I don't think color themes are implemented yet on Android. At least if you are reffering to the colorStyle json attributes. If you mean the AppTheme we haven't gotten all the colors into that yet either.

Reply:
Yeah i meant the color style json attributes and how they hook into the UI